packages/manager/apps/key-management-service/src/components/credential/credentialCreationMethod/credentialCreationMethod.spec.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import { describe, expect, test } from 'vitest';
+import CredentialCreationMethod from './credentialCreationMethod.component';
+
+describe('CredentialStatus component test suite', () => {
+  const useCases: {
+    fromCSR: boolean;
+    label: string;
+  }[] = [
+    {
+      fromCSR: true,
+      label: 'key_management_service_credential_created_with_csr',
+    },
+    {
+      fromCSR: false,
+      label: 'key_management_service_credential_created_without_csr',
+    },
+  ];
+
+  test.each(useCases)(
+    'should return the right translation text for `fromCSR===$fromCSR`',
+    ({ fromCSR, label }) => {
+      // given fromCSR
+
+      // when
+      const { getByText } = render(
+        <CredentialCreationMethod fromCSR={fromCSR} />,
+      );
+
+      // then
+      expect(getByText(label)).toBeInTheDocument();
```

Comment:
**niptick** : you could test also that the other label doesn't appear in the document